### ISSUE TITLE
A specific input makes the component hang (infinite browser loop)

### ISSUE THREAD

**Issue by cristian-spiescu** [probablyup/markdown-to-jsx#496]
Hi,

I found a specific input that makes the component hang the browser:

```jsx
      <Markdown>{
`
|VRAC_T_1|<small>\`🟩-2\`</small> VRAC|<small>🔵<i>end address of original task</i></small><br /><small>\`🟦-1\`</small> F68|<small>🔵<i>parking of \`AF0374\`</i></small><br /><small>\`🟦-6\`</small> E27|
`
      }</Markdown>
```

The initial input was larger (was a line in a table), but I narrowed it down to the minimal that produces this issue. I also made a [repo](https://github.com/cristian-spiescu/markdown-to-jsx-issue) that reproduces this issue. The above code snippet is in [App.tsx](https://github.com/cristian-spiescu/markdown-to-jsx-issue/blob/main/src/App.tsx).

Any ideas what could be the issue?

Thanks in advance!
Cristian

**Comment by cristian-spiescu** [comment]
I'd like to spend a few moments to try to look into this issue. Could anyone share some pointers? Were there similar things in the past? 

Being an infinite loop, its harder to put a breakpoint to start the work. I'd try to pause the VM to try to identify where this loop happens. Maybe in the compiler/parser part? However, I recall that the browser was completely frozen, so how to work w/ the debugger in this case? But maybe I'm wrong.

**Comment by cristian-spiescu** [comment]
A colleague of mine spent a few moments. We didn't find a solid lead. We suspend this for the moment. She'll share below her findings.

**Comment by rvalzz** [comment]
I've experienced the same thing by passing the following string to the `<Markdown />` component, which maxes out the CPU:

```
 "If set (e.g., sort_by[last_login_at]=desc). `authenticate_until`, `email`, `last_desktop_login_at`, `last_login_at`, `username`, `company`, `name`, `notes`, `site_admin`, `receive_admin_alerts`, `allowed_ips`, `password_validity_days`, `ssl_required` or `not_site_admin`."
```

The markdown in this example is malformed to be fair, but ideally the component would bow out gracefully instead of pinning the CPU.

**Comment by daniela-mateescu** [comment]
Hi, 
Unfortunately, we didn't found the cause or a fix for this issue.

From out investigations we  saw that this component  has some performance  problems for some specific inputs and when that input is large enough it crashes. In our case seemed to be related to <span> and _ characters because we reduced our initial input to:

```
      <Markdown>{
`
|a_a_a|<small></small><small></small><small></small><small></small><small></small><small></small><small></small><small></small>
`
      }</Markdown>
```
and for this input it freezes the app, but if you remove any _ or small tag the application doesn't freeze anymore but uses a lot the cpu.

We found that there are many issues registered that seems to be related to our problem:
https://github.com/probablyup/markdown-to-jsx/issues/396
https://github.com/probablyup/markdown-to-jsx/issues/438
https://github.com/probablyup/markdown-to-jsx/issues/374


We are looking forward for any updates regarding this problem, 

Daniela

**Comment by gsouf** [comment]
Interesting one here too using long base64 encoded image urls.

This one will work:

```
![title](someLongBase64Url)
```

But now, if I only add a title line it will fail into a loop and the browser will report too much recursion error:

```
# hello
![title](someLongBase64Url)
```

**Comment by HZ-labs** [comment]
Our error trace for using multiple long base64 encoded image urls:

![error-state](https://github.com/probablyup/markdown-to-jsx/assets/1663933/4d3c7a21-33de-43c3-b995-996e99c1ca5e)

**Comment by cristian-spiescu** [comment]
In my opinion it's a design flow. Not a small issue. We changed to [react-markdown](https://github.com/remarkjs/react-markdown). Seems to be a very solid component.

**Comment by gsouf** [comment]
Yes same here, we did move to `@khanacademy/simple-markdown` that is not well documented, but works well for simple needs and is extensible

**Comment by jackyef** [comment]
Recently encountered this issue. If it helps, the following input cause a high CPU load. It doesn't hang indefinitely, but it takes a long time to finish. (Try pasting it to https://probablyup.com/markdown-to-jsx/)

```
Some text 
Lorem ipsum
Some other text
                 Foo bar
Foo bar baz
                 123-456-1234
                                                                           2 USD.
```

**Comment by CSFelix** [comment]
> Hi, Unfortunately, we didn't found the cause or a fix for this issue.
> 
> From out investigations we saw that this component has some performance problems for some specific inputs and when that input is large enough it crashes. In our case seemed to be related to `<span>` and `_` characters because we reduced our initial input to:
> 
> ```
>       <Markdown>{
> `
> |a_a_a|<small></small><small></small><small></small><small></small><small></small><small></small><small></small><small></small>
> `
>       }</Markdown>
> ```
> 
> and for this input it freezes the app, but if you remove any `_` or `<small>` tag the application doesn't freeze anymore but uses a lot the cpu.
> 
> We found that there are many issues registered that seems to be related to our problem: #396 #438 #374
> 
> We are looking forward for any updates regarding this problem,
> 
> Daniela

Replacing all underscores (_) by anything else solves the issue, but replacing is not a good option since the URL will not be the same anymore.

**Comment by quantizor** [comment]
This is fixed in v9
